homeassistant/components/husqvarna_automower_ble/__init__.py

```diff
@@ -22,10 +22,14 @@
 
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up Husqvarna Autoconnect Bluetooth from a config entry."""
+    if CONF_PIN not in entry.data:
+        raise ConfigEntryAuthFailed
+
     address = entry.data[CONF_ADDRESS]
+    pin = int(entry.data[CONF_PIN])
```

Comment:
Why do we convert to int, don't we store it as a number in the config entry data? If it's for the type checker, we should do a type cast:
```suggestion
    pin = cast(int, entry.data[CONF_PIN])
```

Reply:
It's stored as a string and then we cast it to an int.

Otherwise the Python checkers complain

Reply:
I don't understand this answer. Why do we store it as a string?

Reply:
First of all, you should look into updating the form schemas to use selectors instead of the built-in voluptuous validators, you then get much better control over how the form is presented to the user.
In this case, I would suggest to look into using a `TextSelector` with the mode set to `TextSelectorType.NUMBER`.

Even with a `TextSelector` in number mode, I think you need to validate the input; maybe it's enough to pass the number to your library and check if you can connect to the mower with the supplied PIN.

Reply:
I'm now using `TextSelectorType.NUMBER` and passing the number to the Mower and checking errors from the Mower. I'm not sure there is too much else I can do, as there is a large range of Mowers it isn't obvious what is and isn't a valid PIN

Reply:
Since you convert it to int here, it means you are sure the PIN is always a number. That conversion should then happen in the config flow, and we should store it as a number IMO.